### ISSUE TITLE
contrastive loss

### ISSUE THREAD

**Issue by hamma95** [radekosmulski/whale#17]
I tried changing the loss function to contrastive loss. this is the contrastive loss code:
class ContrastiveLoss(torch.nn.Module):
    """
    Contrastive loss function.
    Based on: http://yann.lecun.com/exdb/publis/pdf/hadsell-chopra-lecun-06.pdf
    """

    def __init__(self, margin=2.0):
        super(ContrastiveLoss, self).__init__()
        self.margin = margin

    def forward(self, output,label):
        euclidean_distance = F.pairwise_distance(output[0], output[1])
        loss_contrastive = torch.mean((1-label) * torch.pow(euclidean_distance, 2) +
                                      (label) * torch.pow(torch.clamp(self.margin - euclidean_distance, min=0.0), 2))

        return loss_contrastive
but i got this error when doing learn.lr_find():
Dimension out of range (expected to be in range of [-1, 0], but got 1)
